theme/snippets/image-caption-module.liquid

```diff
@@ -0,0 +1,37 @@
+{% assign image_module_images = shop.metafields.globals['image_module_image'][module.index] %}
+{% assign caption = shop.metafields.globals['image_module_caption'][module.index] %}
+{% assign variant = shop.metafields.globals['image_module_variant'][module.index] %}
+{% assign color = {% if styling == 'black-bg' %}'color-white'{% else %} 'color-black'{% endif %} %}
+
+{% for image in image_module_images %} 
+  {% assign image_module_image = image %}
+{% endfor %}
+
+{% if image_module_image and variant == 'centered' %}
+  <div class="ImageCaptionModule flex flex-col col-12 md:col-6 mxauto py5 md:py10">
+    <img class="w100 h100 fit-contain" src="{{ image_module_image.src }}" alt="{{ image_module_image.alt }}"/>
+    {% if caption %}
+      <span class="{{ color }} pt_25 sans-xs">{{ caption }}</span>
+    {% endif %}
+  </div>
```

Comment:
It seems like only `ImageCaptionModule` class name changes by the variant. I think you can just do 
```
{% if image_module_image %}
  <div class="ImageCaptionModule flex flex-col 
  {% if variant == 'centered' %}
    col-12 md:col-6 mxauto py5 md:py10
  {% else if variant == 'full-width' %}
    col-12 py3
  {% else if variant == 'left' %}
    sm:flex-row py3
  {% endif %}
">
    <img class="w100 h100 fit-contain" src="{{ image_module_image.src }}" alt="{{ image_module_image.alt }}"/>
    {% if caption %}
      <span class="{{ color }} pt_25 sans-xs">{{ caption }}</span>
    {% endif %}
  </div>

```
something like this (I am not sure that is exact syntax). In that way, we can remove some duplication?

Reply:
thanks Sohee, let me try this!!!

Reply:
Worked! 😄